src/twisted/persisted/aot.py

```diff
@@ -231,13 +231,19 @@ def indentify(s):
     out = []
     stack = []
     l = ["", s]
+
+    def readline():
+        # tokenize no longer supports null bytes in source code since 3.12
+        # see https://github.com/python/cpython/issues/102856
```

Comment:
Maybe the the info as doctring.

I would include 3.11 in the docstrin so that once we remove the support for 3.11 we can refactor this code... or update the comment.

```suggestion
    def pop_line():
        """
        In Python 3.11 or older, tokenize no longer supports null bytes in source code
        # See https://github.com/python/cpython/issues/102856
        """
```

Reply:
This breaks indentify because the null bytes are used as placeholders for indention levels, this is relying on broken behaviour from py3.11. I've vendored the tokenize py3.11 module to fix it